### DISCUSSION TITLE
😽 kitty-scrollback.nvim: Open your Kitty scrollback buffer with Neovim. Ameowzing!

### DISCUSSION THREAD

**Discussion by mikesmithgh** [kovidgoyal/kitty#6637] [Show and tell]
I just released `v1.0.0` of [kitty-scrollback.nvim](https://github.com/mikesmithgh/kitty-scrollback.nvim). If you have any feedback, issues, or suggestions feel free to open an [issue](https://github.com/mikesmithgh/kitty-scrollback.nvim/issues). I'm interested to get feedback if the setup easy or if there are any unclear/confusing steps. This requires Neovim nightly (`v0.10`) and Kitty `v0.29`.

![kitty-scrollback.nvim](https://github.com/kovidgoyal/kitty/assets/10135646/9727d88e-ed44-448b-9ccd-ca371dbb9092)

✨ **Features**

* 😻 Navigate Kitty's scrollback buffer with Neovim
* 🐱 Copy contents from Neovim to system clipboard
* 😺 Send contents from Neovim to Kitty shell
* 🙀 Execute shell command from Neovim to Kitty shell

**Comment by kovidgoyal** [comment]
Added to kitty docs: https://github.com/kovidgoyal/kitty/commit/35766a1a86559cad76195a93c7683b2dfe7c5089

**Reply by mikesmithgh** [reply]
Awesome, thanks!

**Comment by kovidgoyal** [comment]
I looked into the implementation of this, its not clear to me why you need remote control? You can instead use something like this:

```
map ctrl+shift+h launch --type=overlay --stdin-source @screen_scrollback your-kitten.sh
```

your kitten.sh can save the stdin to a file and run neovim on it. Similarly for ctrl+shift+g you use `@last_cmd_output`

Indeed, it might even be possible to just set scrollback_pager to your kitten.sh though that might have knock-on side-effects.

If you do want to use remote control there is no need to allow it globally, you can use launch --allow-remote-control to just allow remote control in the pager window. Be aware that neovim doesnt let you run commands with access to the underlying tty however, so you would need to run your commands to get the output before starting neovim.

**Reply by mikesmithgh** [reply]
Hi @kovidgoyal thanks for the feedback! I originally went that route, however, I ran into some issues passing stdin to Neovim. In order to make this work fast, I pass the stdin to Neovim's builtin terminal. In the case of `@screen_scrollback` the stdin would open in Neovim's buffer. I could do this and read from the buffer to Neovim's terminal, but this approach was slow. I also have some other API calls to Kitty here https://github.com/mikesmithgh/kitty-scrollback.nvim/blob/main/lua/kitty-scrollback/kitty_commands.lua which allow me to run commands in the original window.

That is a good point about the `--allow-remote-control` flag, I can look into that 👍. I was able to workaround the TTY issue with Neovim by using a domain socket.

**Comment by kovidgoyal** [comment]
On Mon, Sep 18, 2023 at 10:26:59AM -0700, Mike wrote:
> Hi @kovidgoyal thanks for the feedback! I originally went that route, however, I ran into some issues passing stdin to Neovim. In order to make this work fast, I pass the stdin to Neovim's builtin terminal. In the case of ***@***.***_scrollback` the stdin would open in Neovim's buffer. I could do this and read from the buffer to Neovim's terminal, but this approach was slow. I also have some other API calls to Kitty here https://github.com/mikesmithgh/kitty-scrollback.nvim/blob/main/lua/kitty-scrollback/kitty_commands.lua which allow me to run commands in the original window.

Ah, OK.

> 
> That is a good point about the `--allow-remote-control` flag, I can look into that 👍. I was able to workaround the TTY issue with Neovim by using a domain socket.

Sadly --allow-remote-control only works via the TTY not a socket, the
reason being that it is window specific and each window is associated
with a tty so using the tty is natural.
